puma/hlplots/results.py

```diff
@@ -115,6 +115,23 @@ def add_taggers_from_file(  # pylint: disable=R0913
         perf_var : np.ndarray, optional
             Override the performance variable to use, by default None
         """
+
+        def filter_nan(data: np.ndarray) -> np.ndarray:
+            """
+            Filter out NaN values from loaded data.
+
+            Parameters
+            ----------
+            data : ndarray
+                Data to filter
+            """
+            mask = np.ones(len(data), dtype=bool)
+            for key in data.dtype.names:
+                mask = np.logical_and(mask, ~np.isnan(data[key]))
```

Comment:
Do we need a loop here? I would have hoped we could do `mask = np.isnan(data).any(-1)` or similar

Reply:
Raw `np.isnan` doesn't work with structured arrays. Converting it to unstructured 4 times slower than the loop :(

Reply:
Ouch. Thanks for the check